spacy/errors.py

```diff
@@ -502,10 +550,12 @@ class Errors:
             "into {values}, but found {value}.")
     E983 = ("Invalid key for '{dict}': {key}. Available keys: "
             "{keys}")
-    E985 = ("The pipeline component '{component}' is already available in the base "
-            "model. The settings in the component block in the config file are "
-            "being ignored. If you want to replace this component instead, set "
-            "'replace' to True in the training configuration.")
+    E984 = ("Invalid component config for '{name}': no @factories key "
+            "specifying the registered function used to initialize the "
+            "component. For example, @factories = \"ner\" will use the 'ner' "
+            "factory and all other settings in the block will be passed "
+            "to it as arguments.\n\n{config}")
+    E985 = ("Can't load model from config file: no 'nlp' section found.\n\n{config}")
```

Comment:
I'm not sure about all these `\n\n{config}` lines in the error messages. I guess it could be useful for debugging, but I'm not a huge fan of spamming the terminal with such a blob of text...

Reply:
Also, the same messages get used when running `debug config`, at which point surely the user knows which config they are debugging and it feels annoying to see it printed again...

Reply:
Maybe we can truncate them? I do think for the smaller sub-blocks, it can be useful, because the origin of the config may not be immediately obvious to the user. It's one thing if you're loading in a `config.cfg` you wrote, but if you have an error somewhere down the line when creating/adding components, I think it makes sense to show the data the error is based on.